src/set/api.rs

```diff
@@ -0,0 +1,706 @@
+		self.inner
+			.iter_mut()
+			.enumerate()
+			.for_each(|(idx, mut bit)| {
+				if *bit && !f(idx) {
+					bit.set(false);
+				}
+			});
+	}
+
+	/// Moves all elements from `other` into `self`, leaving `other` empty.
+	///
+	/// ## Original
+	///
+	/// [`BTreeSet::append`](alloc::collections::BTreeSet::append)
+	///
+	/// ## Examples
+	///
+	/// ```rust
+	/// use bitvec::prelude::*;
+	///
+	/// let mut a = BitSet::new();
+	/// a.insert(1);
+	/// a.insert(2);
+	/// a.insert(3);
+	///
+	/// let mut b = BitSet::new();
+	/// b.insert(3);
+	/// b.insert(4);
+	/// b.insert(5);
+	///
+	/// a.append(&mut b);
+	///
+	/// assert_eq!(a.len(), 5);
+	/// assert_eq!(b.len(), 0);
+	///
+	/// assert!(a.contains(1));
+	/// assert!(a.contains(2));
+	/// assert!(a.contains(3));
+	/// assert!(a.contains(4));
+	/// assert!(a.contains(5));
+	/// ```
+	#[inline]
+	pub fn append(&mut self, other: &mut BitSet<T, O>) {
+		other.shrink_inner();
+		let and_len = Ord::min(self.inner.len(), other.inner.len());
+		self.inner[.. and_len] |= &other.inner[.. and_len];
+		self.inner.extend_from_bitslice(&other.inner[and_len ..]);
+		other.inner.clear();
+	}
+
+	/// Splits the collection into two at the value.
+	/// Returns a new collection with all elements greater than or equal to the
+	/// value.
+	///
+	/// ## Original
+	///
+	/// [`BTreeSet::split_off`](alloc::collections::BTreeSet::split_off)
+	///
+	/// ## Examples
+	///
+	/// ```rust
+	/// use bitvec::prelude::*;
+	///
+	/// let mut a = BitSet::new();
+	/// a.insert(1);
+	/// a.insert(2);
+	/// a.insert(3);
+	/// a.insert(17);
+	/// a.insert(41);
+	///
+	/// let mut b = a.split_off(3);
+	///
+	/// assert_eq!(a.len(), 2);
+	/// assert_eq!(b.len(), 3);
+	///
+	/// assert!(a.contains(1));
+	/// assert!(a.contains(2));
+	/// assert!(b.contains(3));
+	/// assert!(b.contains(17));
+	/// assert!(b.contains(41));
+	/// ```
+	#[inline]
+	pub fn split_off(&mut self, value: usize) -> BitSet<T, O> {
+		let len = self.inner.len();
+		if value > len {
+			self.shrink_inner();
+			return BitSet::new();
+		}
+
+		let mut other = <BitVec<T, O>>::with_capacity(len);
+		unsafe {
+			other.set_len(len);
+			other[.. value].fill(false);
+			other[value ..].copy_from_bitslice(&self.inner[.. value]);
+		}
+		self.inner.truncate(value);
+		self.shrink_inner();
+		BitSet { inner: other }
+	}
+
+	/// Gets an iterator that visits the elements in the bit-set in ascending
+	/// order.
+	///
+	/// ## Original
+	///
+	/// [`BTreeSet::iter`](alloc::collections::BTreeSet::iter)
+	///
+	/// ## Examples
+	///
+	/// ```rust
+	/// use bitvec::prelude::*;
+	///
+	/// let mut set = BitSet::new();
+	/// set.insert(3);
+	/// set.insert(1);
+	/// set.insert(2);
+	///
+	/// let mut set_iter = set.iter();
+	/// assert_eq!(set_iter.next(), Some(1));
+	/// assert_eq!(set_iter.next(), Some(2));
+	/// assert_eq!(set_iter.next(), Some(3));
+	/// assert_eq!(set_iter.next(), None);
+	/// ```
+	#[inline]
+	pub fn iter(&self) -> IterOnes<'_, T, O> {
```

Comment:
I could change this to a wrapper if desired, but I figured reusing the original type was the best idea since any users here should be clear what API is being used.